src/common/core.py

```diff
@@ -685,12 +693,7 @@ def eval(self, x):
             Two dimensional n x m matrix containing the ordinate values 
             corresponding to the argument x.
         """
-        try:
-            y = np.array(np.matrix(self.traj(float(x))))
-        except TypeError:
-            y = np.array(np.matrix(self.traj(x)).transpose())
-                                       #In order to guarantee that the
-                                       #return values are on the correct
-                                       #form. May need to be evaluated
-                                       #for speed improvements.
-        return y
+        if hasattr(x, "__iter__"):
+            return np.vstack([self._single_eval(i) for i in x])
+        else:
+            return self._single_eval(x)
```

Comment:
is there any need to split this into a separate function of it is only invoked from here?

Reply:
You mean the new `_single_eval `I added? Strictly speaking, no; but I think it makes it easier to follow what is going on.

Reply:
Is it possible then maybe to rename it? Initially when reading the code I was thinking "okay what is single eval?", and then I look in the function and I see we are simply returning `np.array(self.traj(x))` with some reshaping, so perhaps something could be clarified here? I have no good suggestion here but I believe it would help someone unfamiliar with this part as me.

Reply:
Changed the naming slightly and added a docstring, please have another look if you think this is more understandable now.

Reply:
Looks good!